src/vcpkg/binarycaching.cpp

```diff
@@ -950,25 +1017,40 @@ namespace
                 if (url_paths.empty()) break;
 
                 print2("Attempting to fetch ", url_paths.size(), " packages from GCS.\n");
+                std::vector<Command> jobs;
+                std::vector<size_t> idxs;
                 for (size_t idx = 0; idx < url_paths.size(); ++idx)
                 {
                     auto&& action = actions[url_indices[idx]];
                     auto&& url_path = url_paths[idx];
                     if (!gsutil_download_file(url_path.first, url_path.second)) continue;
-                    if (decompress_archive(paths, paths.package_dir(action.spec), url_path.second).exit_code != 0)
+                    jobs.push_back(decompress_archive_cmd(paths, paths.package_dir(action.spec), url_path.second));
```

Comment:
This is all good.

Separate / future discussion: have you folks considered using something like `expected<T>` (or `std::variant<T, Error>`) to simplify these loops and eliminate the need for the "original index" stuff?

Reply:
We do have `ExpectedT<T,Error>` in the codebase -- what pattern are you suggesting?

The reason for this "Struct of Arrays" vs "Array of Structs" approach is to enable the parallelization function (`cmd_execute_x_parallel`) to be completely unaware of the outer job indexing, associated data, etc.